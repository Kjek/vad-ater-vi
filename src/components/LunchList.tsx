import type { Restaurant } from '@type/lunch-menu';
import ListItem from './ListItem';

interface LunchListProps {
  restaurants: Restaurant[] | undefined;
}

const LunchList = (props: LunchListProps) => {
  const { restaurants } = props;

  return (
    <div className='flex justify-center pb-8'>
      <div className='container flex justify-center px-4'>
        <div id='restaurant-list' className='flex grow'>
          <ul id='restaurant-list-ul' className='flex grow flex-col'>
            {restaurants
              ? restaurants.map((restaurant) => (
                  <ListItem key={restaurant.name} restaurant={restaurant} />
                ))
              : null}
          </ul>
        </div>
      </div>
    </div>
  );
};

export default LunchList;
