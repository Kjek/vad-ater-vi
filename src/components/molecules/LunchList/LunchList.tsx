import ListItem from '@component/molecules/ListItem/ListItem';
import type { Restaurant } from '@type/lunch-menu';

interface LunchListProps {
  restaurants: Restaurant[] | undefined;
}

const LunchList = (props: LunchListProps) => {
  const { restaurants } = props;

  return (
    <div className='mb-14 flex justify-center py-8 md:mb-0 md:pt-0'>
      <div className='container px-4'>
        <div id='restaurant-list'>
          <ul id='restaurant-list-ul' className='flex grow flex-col'>
            {restaurants
              ? restaurants.map((restaurant) => (
                  <ListItem key={restaurant.name} restaurant={restaurant} />
                ))
              : null}
          </ul>
        </div>
      </div>
    </div>
  );
};

export default LunchList;
