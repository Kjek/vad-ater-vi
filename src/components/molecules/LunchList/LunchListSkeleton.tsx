import ListItemSkeleton from '../ListItem/ListItemSkeleton';

const LunchListSkeleton = () => {
  const restaurants = [1, 2, 3, 4];

  return (
    <div
      role='status'
      className='flex animate-pulse justify-center py-8 pb-16 md:py-0'
    >
      <div className='container mb-6 px-4 md:m-0'>
        <div id='restaurant-list'>
          <ul id='restaurant-list-ul'>
            {restaurants
              ? restaurants.map((_, index) => <ListItemSkeleton key={index} />)
              : null}
          </ul>
        </div>
      </div>
    </div>
  );
};

export default LunchListSkeleton;
