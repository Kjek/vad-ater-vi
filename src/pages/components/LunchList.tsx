import type { Restaurant } from '~/types/lunch-menu';
import ListItem from './ListItem';

interface LunchListProps {
  restaurants: Restaurant[] | undefined;
}

const LunchList = (props: LunchListProps) => {
  const restaurants = props.restaurants;

  return (
    <div id='restaurant-list'>
      <ul id='restaurant-list-ul' className='flex flex-col gap-8'>
        {restaurants ? (
          restaurants.map((restaurant) => (
            <li key={restaurant.name}>
              <ListItem restaurant={restaurant} />
            </li>
          ))
        ) : (
          <p className='text-lg text-white'>Loading menu...</p>
        )}
      </ul>
    </div>
  );
};

export default LunchList;
