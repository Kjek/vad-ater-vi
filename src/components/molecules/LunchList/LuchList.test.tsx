import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import LunchList from './LunchList';

describe(LunchList, () => {
  it('renders correctly', () => {
    const { container } = render(
      <LunchList
        restaurants={[
          {
            name: 'Example',
            menu: [{ day: 'Måndag', food: 'Meatballs' }],
            weeklySpecials: [{ type: 'Veckans sallad', food: 'Ceasarsallad' }],
          },
        ]}
      />
    );
    expect(container).toBeInTheDocument();
    expect(container).toBeVisible();
  });

  it('matches snapshot', () => {
    const defaultRender = render(
      <LunchList
        restaurants={[
          {
            name: 'Example',
            menu: [{ day: 'Måndag', food: 'Meatballs' }],
            weeklySpecials: [{ type: 'Veckans sallad', food: 'Ceasarsallad' }],
          },
        ]}
      />
    );
    expect(defaultRender).toMatchSnapshot();
  });
});
