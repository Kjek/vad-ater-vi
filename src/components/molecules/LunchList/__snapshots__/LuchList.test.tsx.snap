// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LunchList > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mb-14 flex justify-center py-8 md:mb-0 md:pt-0"
      >
        <div
          class="container px-4"
        >
          <div
            id="restaurant-list"
          >
            <ul
              class="flex grow flex-col"
              id="restaurant-list-ul"
            >
              <li
                class="dark:bg-gray-custom whitespace-pre-line break-words border-b bg-white p-6 shadow-md transition duration-500 last:border-none dark:border-gray-700 dark:shadow-none sm:flex"
              >
                <h2
                  class="pb-4 text-center text-3xl font-bold text-gray-800 dark:text-gray-300 sm:w-1/3 sm:grow sm:pb-0"
                >
                  <a
                    title="Gå till restaurangens hemsida"
                  >
                    Example
                  </a>
                </h2>
                <div
                  class="sm:w-2/3 sm:grow"
                >
                  <ul>
                    <li>
                      <h3
                        class="text-2xl font-bold text-gray-800 dark:text-gray-300"
                      >
                        Fel vid hämtning av menyer gå till restaurangens hemsida istället: 
                      </h3>
                      <a
                        class="text-2xl font-bold text-blue-700 underline hover:no-underline dark:text-gray-500"
                      />
                    </li>
                  </ul>
                  <ul
                    class="pt-6"
                    id="weekly-specials-list"
                  >
                    <li
                      class="whitespace-pre-line  "
                    >
                      <h4
                        class="font-bold text-gray-800 dark:text-gray-300"
                      >
                        Veckans sallad
                      </h4>
                      <h5
                        class="text-gray-500 dark:text-gray-300"
                      >
                        Ceasarsallad
                      </h5>
                    </li>
                  </ul>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mb-14 flex justify-center py-8 md:mb-0 md:pt-0"
    >
      <div
        class="container px-4"
      >
        <div
          id="restaurant-list"
        >
          <ul
            class="flex grow flex-col"
            id="restaurant-list-ul"
          >
            <li
              class="dark:bg-gray-custom whitespace-pre-line break-words border-b bg-white p-6 shadow-md transition duration-500 last:border-none dark:border-gray-700 dark:shadow-none sm:flex"
            >
              <h2
                class="pb-4 text-center text-3xl font-bold text-gray-800 dark:text-gray-300 sm:w-1/3 sm:grow sm:pb-0"
              >
                <a
                  title="Gå till restaurangens hemsida"
                >
                  Example
                </a>
              </h2>
              <div
                class="sm:w-2/3 sm:grow"
              >
                <ul>
                  <li>
                    <h3
                      class="text-2xl font-bold text-gray-800 dark:text-gray-300"
                    >
                      Fel vid hämtning av menyer gå till restaurangens hemsida istället: 
                    </h3>
                    <a
                      class="text-2xl font-bold text-blue-700 underline hover:no-underline dark:text-gray-500"
                    />
                  </li>
                </ul>
                <ul
                  class="pt-6"
                  id="weekly-specials-list"
                >
                  <li
                    class="whitespace-pre-line  "
                  >
                    <h4
                      class="font-bold text-gray-800 dark:text-gray-300"
                    >
                      Veckans sallad
                    </h4>
                    <h5
                      class="text-gray-500 dark:text-gray-300"
                    >
                      Ceasarsallad
                    </h5>
                  </li>
                </ul>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
